File: 프로그래머스/unrated/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.java
class Solution {
    public String solution(String my_string, String overwrite_string, int s) {
        String answer = "";
        String[] my_strArr  = my_string.split("");
        String[] overwrite_strArr = overwrite_string.split("");
        for(int i = 0; i < s; i++){
            answer += my_strArr[i];
        }
        for(int i = 0; i < overwrite_strArr.length; i++){
            answer += overwrite_strArr[i];
        }
        if(answer.length()<my_strArr.length){
            int n = my_strArr.length - answer.length();
            for(int i = answer.length(); i < my_strArr.length; i++){
                answer+=my_strArr[i];
            }
        }
        return answer;
    }
}

File: 프로그래머스/unrated/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.py
def solution(my_string, overwrite_string, s):
    answer = ''
    n = len(my_string[s:]) - len(overwrite_string)
    if n > 0:
        answer = (
            str(my_string[:s])
            + str(overwrite_string[:])
            + str(my_string[s+len(overwrite_string):])
        )
    else:
        answer = str(my_string[:s]) + str(overwrite_string[:])
    return answer